libpod/networking_linux.go

```diff
@@ -150,14 +150,31 @@ func (r *RootlessCNI) Do(toRun func() error) error {
 			}
 		}
 
-		// cni plugins need access to /var and /run
-		runDir := filepath.Join(r.dir, "run")
-		varDir := filepath.Join(r.dir, "var")
+		// cni plugins need access to /var/lib/cni and /run
+		varDir := ""
+		varTarget := "/var/lib/cni"
+		// we can only mount to a target dir which exists, check /var/lib/cni recursively
+		// while we could always use /var there are cases where a user might store the cni
+		// configs under /var/custom and this would break
+		for {
+			if _, err := os.Stat(varTarget); err == nil {
+				varDir = filepath.Join(r.dir, varTarget[1:])
```

Comment:
Suggest that a `strings.TrimPrefix` might be a little better than that array slicing - probably a bit slower, but it's safer (guarantees we don't trim a non-slash character if this code ever changes)